# Cálculo - Tarefa 02

1- Calcule de forma numérica a integral da função $x^3$ avaliada entre os pontos $1$ e $2$. Identifique o número de retângulos que você usou.  
2- Calcule essa área de forma analítica.

In [11]:
import numpy as np

In [15]:
# Função f(x) = x³
f = lambda x: x**3

# Função para calcular a área usando a regra dos retângulos (ponto médio)
def calcula_area(a, b, func, frame_num):
    num_bins = frame_num + 1  # Quantidade de retângulos
    delta = (b - a) / num_bins  # Largura do retângulo
    x_cols = np.linspace(a, b - delta, num_bins) + delta / 2  # Ponto médio
    y_cols = func(x_cols)  # Alturas dos retângulos
    return sum(delta * y_cols)

# Intervalo de integração e número de retângulos
a, b = 1, 2
frame_num = 100  # 100 retângulos

# Calculando a área
area = calcula_area(a, b, f, frame_num)

# Exibindo a área e o número de retângulos usados
print(f"A integral numérica da função x³ entre 1 e 2 é: {area}")
print(f"Número de retângulos usados: {frame_num + 1}")


A integral numérica da função x³ entre 1 e 2 é: 3.7499632388981476
Número de retângulos usados: 101


Calculando a área sob a curva da função x³ entre 1 e 2 de forma analítica.

A integral de $x^3$ é dada por:

$$
\int x^3 \, dx = \frac{x^4}{4} + C
$$


Fica assim:

$$
\int_1^2 x^3 \, dx = \frac{16}{4} - \frac{1}{4} = 4 - 0.25 = 3.75
$$

Portanto, a área sob a curva da função $x^3$ entre 1 e 2 é **3.75**.
